src/scene/materials/standard-material.js

```diff
@@ -382,48 +367,34 @@ class StandardMaterial extends Material {
      */
     clone() {
         const clone = new StandardMaterial();
-        this._cloneInternal(clone);
 
-        for (let i = 0; i < _propsSerial.length; i++) {
-            const pname = _propsSerial[i];
-            if (this[pname] !== undefined) {
-                if (this[pname] && this[pname].copy) {
-                    if (clone[pname]) {
-                        clone[pname].copy(this[pname]);
-                    } else {
-                        clone[pname] = this[pname].clone();
-                    }
-                } else {
-                    clone[pname] = this[pname];
-                }
-            }
-        }
+        this._cloneInternal(clone);
 
-        return clone;
-    }
+        // set properties
+        Object.keys(_props).forEach((k) => {
+            clone[k] = this[k];
+        });
 
-    _updateMapTransform(transform, tiling, offset) {
-        if (tiling.x === 1 && tiling.y === 1 && offset.x === 0 && offset.y === 0) {
-            return null;
+        // clone chunks
+        for (const p in this._chunks) {
+            if (this._chunks.hasOwnProperty(p))
+                clone._chunks[p] = this._chunks[p];
         }
 
-        transform = transform || new Vec4();
-        transform.set(tiling.x, tiling.y, offset.x, 1.0 - tiling.y - offset.y);
-        return transform;
+        return clone;
     }
 
     _setParameter(name, value) {
         if (!this.parameters[name])
-            this._propsSet.push(name);
+            _propsSet.push(name);
         this.setParameter(name, value);
     }
 
     _clearParameters() {
-        const props = this._propsSet;
-        for (let i = 0; i < props.length; i++) {
-            delete this.parameters[props[i]];
-        }
-        this._propsSet = [];
+        _propsSet.forEach((p) => {
+            delete this.parameters[p];
+        });
+        _propsSet = [];
```

Comment:
maybe just set .length to 0 instead of allocating new array

Reply:
yep good call